AmplifyPlugins/Auth/Tests/AuthHostApp/AuthIntegrationTests/CredentialStore/CredentialStoreConfigurationTests.swift

```diff
@@ -210,4 +210,315 @@ class CredentialStoreConfigurationTests: AWSAuthBaseTest {
         let credentials = try? newCredentialStore.retrieveCredential()
         XCTAssertNil(credentials)
     }
+    
+    /// Test migrating to a shared access group keeps credentials
+    ///
+    /// - Given: A user registered is configured
+    /// - When:
+    ///    - The credential store is re-initialized with shared access group and migration set to true
+    /// - Then:
+    ///    - The old credentials should still persist
+    ///
+    func testCredentialsRemainOnMigrationToSharedAccessGroup() {
+        // Given
+        let identityId = "identityId"
+        // Migration only happens if credentials are not expired, hence
+        // the need for nonimmediate expiration test data
+        let awsCredentials = AuthAWSCognitoCredentials.nonimmediateExpiryTestData
+        let initialCognitoCredentials = AmplifyCredentials.userPoolAndIdentityPool(
+            signedInData: .testData,
+            identityID: identityId,
+            credentials: awsCredentials)
+        let initialAuthConfig = AuthConfiguration.userPoolsAndIdentityPools(
+            Defaults.makeDefaultUserPoolConfigData(),
+            Defaults.makeIdentityConfigData())
+        let credentialStore = AWSCognitoAuthCredentialStore(authConfiguration: initialAuthConfig)
+        do {
+            try credentialStore.saveCredential(initialCognitoCredentials)
+        } catch {
+            XCTFail("Unable to save credentials")
+        }
+
+        // When migrating to shared access group with same configuration
+        #if os(watchOS)
+        let newCredentialStore = AWSCognitoAuthCredentialStore(authConfiguration: initialAuthConfig, accessGroup: keychainAccessGroupWatch, migrateKeychainItemsOfUserSession: true)
+        #else
+        let newCredentialStore = AWSCognitoAuthCredentialStore(authConfiguration: initialAuthConfig, accessGroup: keychainAccessGroup, migrateKeychainItemsOfUserSession: true)
+        #endif
+
+        // Then
+        guard let credentials = try? newCredentialStore.retrieveCredential(),
+              case .userPoolAndIdentityPool(let retrievedTokens,
+                                            let retrievedIdentityID,
+                                            let retrievedCredentials) = credentials else {
+            XCTFail("Unable to retrieve Credentials")
+            return
+        }
+        XCTAssertNotNil(credentials)
+        XCTAssertNotNil(retrievedTokens)
```

Comment:
is it possible to assert on the equality of `retrievedTokens` as well?

Reply:
if it's not nil, then it will be the same (the test data is a string that won't change), so I'm not sure it's necessary